documentation-site/pages/guides/theming.mdx

```diff
@@ -6,126 +6,668 @@ LICENSE file in the root directory of this source tree.
+
+### Customizing Icons
+
+The theme object is also the source for customizing the icons we use in Base Web components.
+
+You can use any React component as a replacement for an icon, as long as they accept the following properties:
+
+- `size`, to set their width and height
+- `color`, to set their colors
+- `title`, to set the title of the icon for accessibility purposes
 
 You can find more about the [Icon API here](/components/icon/#icon-api) and also refer to this [list of available icons](https://baseweb.design/components/icon/#available-icons).
 
-The example below overrides the `ChevronLeft` icon with the `ArrowLeft` icon, making the pagination slightly different.
+The example below overrides the `ChevronLeft` icon with the `ArrowLeft` icon, making the pagination appear slightly different.
 
 All the icons that can be found on the Icon component page can be overridden using this technique.
 
 <Example title="Pagination icon overrides" path="theme/icon-overrides.js">
   <IconOverrides />
 </Example>
+
+### Customizing the Theme Type
+
+<Notification
+  kind="warning"
+  overrides={{
+    Body: {
+      style: {
+        width: '100%',
+      },
+    },
+  }}
+>
+  Extending the theme is an advanced pattern. See if you can configure the
+  default theme first before extending it.
+</Notification>
+
+There may be a scenario where you want to extend our default theme. We do not reccomended removing theme properties, as this may lead to a run-time error when a component references a missing property. However, _adding_ new properties is perfectly acceptable.
+
+```js
+// @flow
+
+import type {ThemeT} from 'baseui';
+
+const CustomThemeT = ThemeT & {customColor: string};
+```
+
+You will notice however, that you can't use `createTheme` with your `CustomThemeT`. If you want to use `createTheme`, we reccommend that you first create a normal theme with `createTheme` and then _extend_ it with your custom properties.
+
+```js
+import {createTheme} from 'baseui';
+import type {ThemeT} from 'baseui';
+
+const CustomThemeT = ThemeT & {customColor: string};
+
+const theme = createTheme(/*primtives, overrides*/);
+const customTheme: CustomThemeT = {
+  ...theme,
+  customColor: 'pink',
+};
+
+function App() {
+  return <BaseProvider theme={customTheme}>{/* ... */}</BaseProvider>;
+}
+```
+
+Note, you can use the custom type theme object with our provider so long as it _extends_ the default `ThemeT`. Base Web components will essentially ignore the additional properties, but the theme object will be available in component overrides.
+
+If you want to use that custom theme shape with our various styling utilties, you will want create new `styled`, `withStyle`, and `useStyletron` methods which use your custom theme type. Luckily, we make this pretty easy for you.
```

Comment:
You may want to communicate that this step is only required when using flow. I'm not sure about typescript though, it may be `any`